File: day1.swift
#! /usr/bin/swift

var modules = [Int]()
while let line = readLine() {
  modules.append(Int(line)!)
}

print(modules.reduce(0, { $0 + ($1 / 3) - 2 }))

func calculate_fuel(_ module: Int) -> Int {
  var fuel = (module / 3) - 2
  if fuel > 0 {
    fuel += calculate_fuel(fuel)
  } else {
    fuel = 0
  }
  return fuel
}

print(modules.reduce(0, { $0 + calculate_fuel($1) }))

File: day1.py
#! /usr/bin/env python3

from functools import reduce

modules = list(map(lambda module: int(module.strip()), tuple(open('day1.txt', 'r'))))

print(reduce(lambda memo, m: memo + (int(m / 3.0) - 2), modules, 0))

def calculate_fuel(module):
    fuel = int(module / 3.0) - 2
    if fuel > 0:
        fuel += calculate_fuel(fuel)
    else:
        fuel = 0
    return fuel

print(reduce(lambda memo, module: memo + calculate_fuel(module), modules, 0))
